omfhint.cpp

```diff
@@ -142,7 +149,78 @@ OMFHintFilter::configure(const ConfigCategory& config)
 				} else {
 					m_hints.insert(pair<string, string>(asset, escaped));
 				}
+				// Check if macro substituion is required
+				// At least one pair of '$' sign must be there to apply macro
+				if (std::count(escaped.begin(), escaped.end(), '$') > 1)
+					isMacro = true;
 			}
 		}
+		if (isMacro)
+			collectMacrosInfo(config.getValue("hints"));
+	}
+}
+
+/**
+ * Extract datapoint name for macro replacement
+ *
+ * @param hintsJson	OMFHints JSON
+ */
+void OMFHintFilter::collectMacrosInfo(std::string hintsJSON)
+{
+	std::regex pattern("\\$([^$]*)\\$");
```

Comment:
regex is extremely slow - this will have a very big impact on the performance of the system, although it is only during reconfigure, so not as bad as if you had done this for every reading you process.

Reply:
Removed Regex